clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -1,5 +1,5 @@
 /* **********************************************************
```

Comment:
> Before this PR, as we were pushing the dummy zero entry at kernel_event markers inside the kernel trace with no corresponding pop, 

See prior comment: I don't follow this.  Shouldn't the pop be at kernel_xfer and thus would still happen in kernel trace?  And if kernel_event has a push, why do you say above "We have existing logic that skips pushing and popping from the retaddr stack for the kernel parts"?

Reply:
>  Shouldn't the pop be at kernel_xfer and thus would still happen in kernel trace?

1. pop is not at kernel_xfer at all. It is at return-instr.
2. we have existing logic to skip popping from retaddr stack in the kernel trace. we also have existing logic to skip pushing to the retaddr stack at call-instr in the kernel trace. what was missing was logic to skip pushing to the retaddr stack at kernel_event inside the kernel trace.

Reply:
> why do you say above "We have existing logic that skips pushing and popping from the retaddr stack for the kernel parts"?

Note that there's a "But ..." in the next sentence in the same paragraph that notes the missing handling for kernel_event specifically. I did s/We have existing logic/We have some existing logic/.